Source/RxCocoa/UIScrollView+reachedBottom.swift

```diff
@@ -0,0 +1,30 @@
+//
+//  UIScrollView+reachedBottom.swift
+//  RxSwiftExt
+//
+//  Created by Anton Nazarov on 09/05/2019.
+//  Copyright © 2019 RxSwift Community. All rights reserved.
+//
+
+import UIKit
+import RxSwift
+import RxCocoa
+
+public extension Reactive where Base: UIScrollView {
+    /**
+     Shows if the bottom of the UIScrollView is reached.
+     - parameter offset: A threshhold indicating the bottom of the UIScrollView.
+     - returns: ControlEvent that emits when the bottom of the base UIScrollView is reached.
+     */
+    func reachedBottom(offset: CGFloat = 0.0) -> ControlEvent<Void> {
+        let source = contentOffset.map { contentOffset in
+            let visibleHeight = self.base.frame.height - self.base.contentInset.top - self.base.contentInset.bottom
+            let y = contentOffset.y + self.base.contentInset.top
+            let threshold = max(offset, self.base.contentSize.height - visibleHeight)
+            return y >= threshold
+        }
+        .filter { $0 }
```

Comment:
By the way, not sure if you'd want to but you can replace most of the implementation with a flatMap:

```swift
let source = contentOffset.flatMap { contentOffset -> Observable<Void> in
            let visibleHeight = self.base.frame.height - self.base.contentInset.top - self.base.contentInset.bottom
            let y = contentOffset.y + self.base.contentInset.top
            let threshold = max(offset, self.base.contentSize.height - visibleHeight)
            
            guard y >= threshold else { return .never() }
            return .just(())
        }
```

You don't have to but it's just a suggestion if you want to avoid passing through 3 operators (should be negligible).

Reply:
I saw this option of the implementation. I think it's better to use `map` & `filter` composition here, because it's not a common way of using `flatMap`.

>most of the implementation

I can see one line saving. Not sure it's really big improvement. What do you think?

Reply:
Matter of preference, I don't care at all :) 
There's no problem with using filter + map and there's nothing wrong with using `flatMap` for this as well. Feel free to disregard, I just offered it as an option.

Reply:
Thank you) Good option, but i am ok with current impl